addons/mail/static/src/messaging/entity/notification_group_manager/notification_group_manager.js

```diff
@@ -0,0 +1,73 @@
+odoo.define('mail.messaging.entity.NotificationGroupManager', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entityCore');
+const { one2many } = require('mail.messaging.EntityField');
+
+function NotificationGroupManagerFactory({ Entity }) {
+
+    class NotificationGroupManager extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        computeGroups() {
+            if (!this.env.messaging.currentPartner) {
+                this.update({ groups: [['unlink-all']] });
+                return;
+            }
+            const groups = [];
+            this.env.messaging.currentPartner.failureNotifications.forEach(notification => {
+                const thread = notification.message.originThread;
+                // Notifications are grouped by model and notification_type.
+                // Except for channel where they are also grouped by id because
+                // we want to open the actual channel in discuss or chat window
+                // and not its kanban/list/form view.
+                const channelId = thread.model === 'mail.channel' ? thread.id : null;
+                const id = `${thread.model}/${channelId}/${notification.notification_type}`;
+                const group = this.env.entities.NotificationGroup.insert({
+                    id,
+                    model: thread.model,
+                    model_name: thread.model_name,
+                    notification_type: notification.notification_type,
+                });
+                group.update({ notifications: [['link', notification]] });
```

Comment:
note that this reuses a notification group and just append notification. `computeGroups()` doesn't seem to handle case of a notification group that should unlink some notifications

Reply:
It's not a case that happens in practice but I guess it's safer to handle it too.